java/Dockerfile

```diff
@@ -15,16 +15,11 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM maven
+FROM maven:3.6-jdk-8
 
-ADD header /arrow/
-ADD format /arrow/format
-ADD integration/data /arrow/integration/data
-ADD java /arrow/java
 WORKDIR /arrow/java
 
-# build
-RUN mvn -DskipTests=true -Dcheckstyle.skip=true -B install
-
-# test
-CMD mvn test
+CMD mvn -Dcheckstyle.skip=true \
+        -Dmaven.repo.local=/build/java/repository \
```

Comment:
So I use the mounted `arrow/docker_cache:/build` directory to cache as much as possible. It already spares the maven package downloading step, but it'd be nice to cache `**/target` directories as well, can We do that without to much maven magic?